// The Firmament project
// Copyright (c) 2011-2012 Malte Schwarzkopf <malte.schwarzkopf@cl.cam.ac.uk>
//
// HeartbeatMessage is a simple keep-alive message that reports on a live
// resource. It may also trigger registration and fault recovery, although
// initial registration is triggered explicitly using RegistrationMessage.

syntax = "proto3";

package firmament;

import "base/resource_desc.proto";
import "base/machine_perf_statistics_sample.proto";

message HeartbeatMessage {
  string uuid = 1;
  string location = 2;
  uint64 capacity = 3;
  MachinePerfStatisticsSample load = 4;
  ResourceDescriptor res_desc = 5;
}
